Core/Object Arts/Dolphin/Base/Integer.cls

```diff
@@ -497,6 +497,23 @@ isPowerOfTwo
 
 	^self strictlyPositive and: [(self bitAnd: self - 1) = 0]!
 
+isPrime
+	"Answer true if the receiver is a prime number."
+
+	| maxDivider |
+
+	"Handle special cases."
+	self < 2 ifTrue: [ ^ false ].					"1, 0 and negative numbers are not prime."
+	( self = 2 or: [ self = 3 ] ) ifTrue: [ ^ true ].	"2 and 3 are prime."
+	self \\ 2 = 0 ifTrue: [ ^ false ].				"Other even numbers are not prime."
+
+	"Now check divisibility by all odd dividers upto the square root of self."
+	maxDivider := self sqrt asInteger + 1.
```

Comment:
One other thought: `#sqrt` is inexact for integers that are too large to be represented accurately as a 64-bit Float, i.e. any integer > `16r20000000000000`. I'm not sure exactly what effect that might have here without further thought/testing.